storybookjs/storybook
# Addon-docs: Refactor docs support into individual framework packages

Issue: #16210

## What I did

- [x] Created a new library `@storybook/docs-tools` to contain common functions
- [x] Moved framework into `angular`, `ember`, `html`, `react`, `svelte`, `vue`, `vue`, `web-components`
- [x] Cleaned up `addon-docs` complex preset system 

In the telescoping PR https://github.com/storybookjs/storybook/pull/17697
- [x] Added the ability to disable docs support for these packages

## What to review

1. Naming: does `docs-tools` make sense?
2. Overall structure: does the framework update with the `preset-framework-docs`, which in turn calls into `docs-tools` make sense? Does this bloat the framework packages in some way that I'm missing, other than what's handled in #17697?

## How to test

I tested this using Storybook test-runner, doing visual snapshot tests against the docs pages in both `next` and this branch.

You can see the feature/instructions here: https://github.com/storybookjs/test-runner/pull/79
